server/src/main/java/org/opensearch/index/translog/Checkpoint.java

```diff
@@ -207,9 +213,8 @@ public static void write(ChannelFactory factory, Path checkpointFile, Checkpoint
         // now go and write to the channel, in one go.
         try (FileChannel channel = factory.open(checkpointFile, options)) {
             Channels.writeToChannel(bytes, channel);
-            // no need to force metadata, file size stays the same and we did the full fsync
-            // when we first created the file, so the directory entry doesn't change as well
-            channel.force(false);
+            // force fsync with metadata since this is used on file creation
+            channel.force(true);
```

Comment:
Basically made all `IOUtils.fsync`  calls redundant but any performance impact?

Reply:
We could do this as a separate PR to measure performance though I don't expect any since the fsyncs were redundant anyway?

Really this method is only called from `Translog#createEmptyTranslog`, `TranslogReader#closeIntoTrimmedReader`, and `TruncateTranslogAction#writeEmptyCheckpoint` which is used sparingly.

Reply:
:+1: